### PR TITLE
Spelling siloptimizer

### PR DESCRIPTION
**Pull request by jsoref**
This fixes some misspellings in Swift siloptimizer, it is split per https://github.com/apple/swift/pull/42421#issuecomment-1101092698

### CODE REVIEW

**Comment by tshortli** [comment]
@swift-ci please smoke test

File: test/SILOptimizer/functionsigopts.sil [hunk 1]
```diff
@@ -1781,27 +1781,27 @@ bb0(%0 : $T):
   return %r : $()
 }
 
-// CHECK-LABEL: sil [signature_optimized_thunk] [always_inline] @generic_func_with_multple_generic_args_and_dead_generic_arg : $@convention(thin) <T, S> (@in T, @in S) -> ()
-// CHECK: function_ref @$s026generic_func_with_multple_a15_args_and_dead_A4_argTf4nd_n : $@convention(thin) <τ_0_0, τ_0_1> (@in τ_0_0) -> ()
+// CHECK-LABEL: sil [signature_optimized_thunk] [always_inline] @generic_func_with_multiple_generic_args_and_dead_generic_arg : $@convention(thin) <T, S> (@in T, @in S) -> ()
+// CHECK: function_ref @$s026generic_func_with_multiple_a15_args_and_dead_A4_argTf4nd_n : $@convention(thin) <τ_0_0, τ_0_1> (@in τ_0_0) -> ()
```
Comment (jsoref) [inline test/SILOptimizer/functionsigopts.sil:1785]: > Failed Tests (1):
>  Swift(macosx-x86_64) :: SILOptimizer/functionsigopts.sil

I'm going to drop the changes to this file.

**Comment by tshortli** [comment]
@swift-ci please smoke test

**Comment by tshortli** [comment]
@swift-ci please smoke test

**Review by tshortli — APPROVED** [review]
I'm not an expert in this area, but I did review all the changes and they all seems like straightforward typo fixes so LGTM. @jsoref could you squash?

**Comment by jsoref** [comment]
@tshortli done

**Comment by tshortli** [comment]
@swift-ci please smoke test

**Pull request merged**